### ISSUE TITLE
ssh connection shutdown with valid credential

### ISSUE THREAD

**Issue by nikaiw** [Pennyw0rth/NetExec#105]
Using ssh with valid credential on a server led to just EOF

This might have to do with the server using ssh-rsa /dss ? 

```
2023-11-01 19:38:09,509 - DEBUG - Added file handler: <RotatingFileHandler /home/user/.nxc/logs/2023-11-01/log_2023-11-01-19-38-09.log (NOTSET)>
2023-11-01 19:38:09,512 - DEBUG - PYTHON VERSION: 3.11.6 (main, Oct  8 2023, 05:06:43) [GCC 13.2.0]
2023-11-01 19:38:09,513 - DEBUG - RUNNING ON: Linux Release: 6.5.0-10-generic
2023-11-01 19:38:09,514 - DEBUG - Passed args: Namespace(threads=100, timeout=None, jitter=None, no_progress=False, verbose=False, debug=True, version=False, protocol='ssh', target=['<redacted>'], cred_id=[], username=['<redacted>'], password=['<redacted>'], ignore_pw_decoding=False, kerberos=False, no_bruteforce=False, continue_on_success=False, use_kcache=False, log=None, aesKey=None, kdcHost=None, gfail_limit=None, ufail_limit=None, fail_limit=None, module=None, module_options=[], list_modules=False, show_module_options=False, server='https', server_host='0.0.0.0', server_port=None, connectback_host=None, key_file=None, port=22, ssh_timeout=30, sudo_check=False, sudo_check_method='sudo-stdin', get_output_tries=5, codec='utf-8', no_output=False, execute=None)
2023-11-01 19:38:09,517 - DEBUG - Protocol: ssh
2023-11-01 19:38:09,518 - DEBUG - Protocol Path: /home/user/tools/NetExec/venv/lib/python3.11/site-packages/nxc/protocols/ssh.py
2023-11-01 19:38:09,519 - DEBUG - Protocol DB Path: /home/user/tools/NetExec/venv/lib/python3.11/site-packages/nxc/protocols/ssh/database.py
2023-11-01 19:38:09,672 - DEBUG - Protocol Object: <class 'protocol.ssh'>
2023-11-01 19:38:09,674 - DEBUG - Protocol DB Object: <class 'protocol.database'>
2023-11-01 19:38:09,675 - DEBUG - DB Path: /home/user/.nxc/workspaces/default/ssh.db
2023-11-01 19:38:09,689 - DEBUG - Creating ThreadPoolExecutor
2023-11-01 19:38:09,690 - DEBUG - Creating thread for <class 'protocol.ssh'>
2023-11-01 19:38:09,692 - INFO - Socket info: host=<redacted>, hostname=<redacted>, kerberos=False
2023-11-01 19:38:09,693 - DEBUG - Kicking off proto_flow
2023-11-01 19:38:09,853 - DEBUG - Remote version: SSH-2.0-OpenSSH_5.3
2023-11-01 19:38:09,856 - DEBUG - add_host(): Initial hosts results: [(19, '<redacted>', 22, 'SSH-2.0-OpenSSH_5.3', '')]
2023-11-01 19:38:09,857 - DEBUG - host: (19, '<redacted>', 22, 'SSH-2.0-OpenSSH_5.3', '')
2023-11-01 19:38:09,858 - DEBUG - host_data: {'id': 19, 'host': '<redacted>', 'port': 22, 'banner': 'SSH-2.0-OpenSSH_5.3', 'os': ''}
2023-11-01 19:38:09,859 - DEBUG - Hosts: [{'id': 19, 'host': '<redacted>', 'port': 22, 'banner': 'SSH-2.0-OpenSSH_5.3', 'os': ''}]
2023-11-01 19:38:09,861 - DEBUG - add_host() - Host IDs Updated: [19]
2023-11-01 19:38:09,897 - INFO - SSH         <redacted>   22     <redacted>    [*] SSH-2.0-OpenSSH_5.3
2023-11-01 19:38:09,898 - DEBUG - Trying to authenticate using plaintext over SSH
2023-11-01 19:38:09,899 - DEBUG - Logging <redacted> with username: <redacted>, password: <redacted>
2023-11-01 19:38:09,911 - INFO - SSH         <redacted>   22     <redacted>    [-] <redacted>:<redacted> Authentication failed: transport shut down or saw EOF
```

**Comment by XiaoliChan** [comment]
Still have no idea about paramiko XD

**Comment by nikaiw** [comment]
fun fact I ran the test again from another laptop:
```
[18:58:30] DEBUG    PYTHON VERSION: 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0]                                                                                                                                          netexec.py:89
           DEBUG    RUNNING ON: Linux Release: 6.2.0-34-generic                                                                                                                                                                 netexec.py:90
           DEBUG    Passed args: Namespace(threads=100, timeout=None, jitter=None, no_progress=False, verbose=False, debug=True, version=False, protocol='ssh', target=['44.249.132.70'], cred_id=[], username=['SA-NA-MD'],    netexec.py:91
                    password=['Airbus@123'], ignore_pw_decoding=False, kerberos=False, no_bruteforce=False, continue_on_success=False, use_kcache=False, log=None, aesKey=None, kdcHost=None, gfail_limit=None,                              
                    ufail_limit=None, fail_limit=None, module=None, module_options=[], list_modules=False, show_module_options=False, server='https', server_host='0.0.0.0', server_port=None, connectback_host=None,                        
                    key_file=None, port=22, ssh_timeout=15, sudo_check=False, sudo_check_method='sudo-stdin', get_output_tries=5, codec='utf-8', no_output=False, execute=None)                                                              
           DEBUG    Protocol: ssh                                                                                                                                                                                              netexec.py:145
           DEBUG    Protocol Path: /home/user/tools/NetExec/nxc/protocols/ssh.py                                                                                                                                               netexec.py:148
           DEBUG    Protocol DB Path: /home/user/tools/NetExec/nxc/protocols/ssh/database.py                                                                                                                                   netexec.py:150
           DEBUG    Protocol Object: <class 'protocol.ssh'>                                                                                                                                                                    netexec.py:153
           DEBUG    Protocol DB Object: <class 'protocol.database'>                                                                                                                                                            netexec.py:155
           DEBUG    DB Path: /home/user/.nxc/workspaces/default/ssh.db                                                                                                                                                         netexec.py:158
           DEBUG    Using selector: EpollSelector                                                                                                                                                                       selector_events.py:54
           DEBUG    Creating ThreadPoolExecutor                                                                                                                                                                                 netexec.py:47
           DEBUG    Creating thread for <class 'protocol.ssh'>                      
```

That time however it worked . Both are fresh install of netexec running in their dedicated virtualenv

**Comment by nikaiw** [comment]
Hm so issue is different than expected, actually paramiko try to use ssh agent by default. So even when provided 
```-u user -p password``` it was first trying to auth using the ssh key. Auth works and is correctly reported by netexec after doing ```unset SSH_AUTH_SOCK```

**Comment by XiaoliChan** [comment]
@NeffIsBack This issue can close now

**Comment by NeffIsBack** [comment]
Closing as fixed in #106
